lms/djangoapps/shoppingcart/migrations/0026_migrate_invoices.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+from decimal import Decimal, ROUND_DOWN
+from south.v2 import DataMigration
+
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        "Write your forwards methods here."
+        # Note: Remember to use orm['appname.ModelName'] rather than "from appname.models..."
+        for invoice in orm.Invoice.objects.extra(
+            select={
+                'qty': 'SELECT COUNT(*) FROM shoppingcart_courseregistrationcode '
+                       'WHERE shoppingcart_courseregistrationcode.invoice_id = shoppingcart_invoice.id'
+            }
+        ).all():
+            invoice_item = self._create_invoice_item(invoice, orm)
+            orm.CourseRegistrationCode.objects.filter(invoice=invoice).update(invoice_item=invoice_item)
+
+    def backwards(self, orm):
+        "Write your backwards methods here."
+
+    @staticmethod
+    def _create_invoice_item(invoice, orm):
+        unit_price = ((Decimal(invoice.total_amount))/invoice.qty).quantize(Decimal('.01'), rounding=ROUND_DOWN)
```

Comment:
Especially if we preserve the old `total_amount` field and default to that if it's set, as described above.